noria-server/src/controller/sql/mir/mod.rs

```diff
@@ -199,25 +233,34 @@ impl SqlToMirConverter {
                 FilterCondition::Comparison(ct.operator.clone(), filter::Value::Column(fi))
             }
             _ => unimplemented!(),
-        });
+        };
 
         let absolute_column_ids: Vec<usize> = columns
             .iter()
             .map(|c| n.borrow().column_id_for_column(c, None))
             .collect();
         let max_column_id = *absolute_column_ids.iter().max().unwrap();
         let num_columns = max(columns.len(), max_column_id + 1);
-        let mut filters = vec![None; num_columns];
+        let mut filters = Vec::new();
 
         match columns.iter().rposition(|c| *c.name == l.name) {
             None => {
                 // Might occur if the column doesn't exist in the parent; e.g., for aggregations.
                 // We assume that the column is appended at the end.
-                columns.push(Column::from(l));
-                filters.push(f);
+                match n.borrow().inner {
+                    MirNodeType::Aggregation { .. }  => {
+                        columns.insert(columns.len()-1, Column::from(l));
```

Comment:
Please update comment above to say that we append the column at the end unless we have and aggregation, in which case it needs to go before the computed column, which is last.